adapters/ix/ix.go

```diff
@@ -359,3 +363,33 @@ func BuildIxDiag(request *openrtb2.BidRequest) error {
 	}
 	return nil
 }
+
+// moves sid from imp[].ext.bidder.sid to imp[].ext.sid
+func moveSid(imps []openrtb2.Imp) error {
+	if len(imps) < 1 {
+		return nil
+	}
+	imp := imps[0]
+	if imp.Ext == nil || len(imp.Ext) == 0 {
+		return nil
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return err
+	}
+
+	var ixExt openrtb_ext.ExtImpIx
+	if err := json.Unmarshal(bidderExt.Bidder, &ixExt); err != nil {
+		return err
+	}
+
+	if ixExt.Sid != "" {
+		var m map[string]interface{}
+		json.Unmarshal(imp.Ext, &m)
+		m["sid"] = ixExt.Sid
+		ext, _ := json.Marshal(m)
```

Comment:
You should not silently ignore marshal error. It should be - 
```
ext, err := json.Marshal(m)
if err != nil {
     //continue or return error // depends on what you want to do.
}
```

Reply:
Agreed. There are sometimes shared memory conflicts which will cause a marshal error. We should allow this error to bubble up and be caught by the main bidder loop.